README.md

```diff
@@ -277,6 +277,7 @@ See code for all available configurations.
 | [Purism Librem 5r4](purism/librem/5r4)                                 | `<nixos-hardware/purism/librem/5r4>`                    |
 | [Raspberry Pi 2](raspberry-pi/2)                                       | `<nixos-hardware/raspberry-pi/2>`                       |
 | [Raspberry Pi 4](raspberry-pi/4)                                       | `<nixos-hardware/raspberry-pi/4>`                       |
+| [Raspberry Pi 5](raspberry-pi/5)                                       | `<nixos-hardware/raspberry-pi/5>`                       |
```

Comment:
an entry in `flake.nix` is missing to be usable like described in the README.md:
```nix
imports = [
    nixos-hardware.nixosModules.raspberry-pi-5

   ];
```

Reply:
Hm, i'm getting:

```shell
nix build .\#nixosConfigurations.pi5.config.system.build.toplevel
error: builder for '/nix/store/53vdydgf8mcjvf8w4g42agd6cpz4m715-linux-config-aarch64-unknown-linux-gnu-6.1.21-1.20230405.drv' failed with exit code 2;
       last 10 log lines:
       >   HOSTCC  scripts/kconfig/symbol.o
       >   HOSTCC  scripts/kconfig/util.o
       >   HOSTLD  scripts/kconfig/conf
       > ***
       > *** Can't find default configuration "arch/arm64/configs/bcm2712_defconfig"!
       > ***
       > make[2]: *** [../scripts/kconfig/Makefile:94: bcm2712_defconfig] Error 1
       > make[1]: *** [/build/source/Makefile:704: bcm2712_defconfig] Error 2
       > make: *** [Makefile:238: __sub-make] Error 2
       > make: Leaving directory '/build/source'
       For full logs, run 'nix log /nix/store/53vdydgf8mcjvf8w4g42agd6cpz4m715-linux-config-aarch64-unknown-linux-gnu-6.1.21-1.20230405.drv'.
error: 1 dependencies of derivation '/nix/store/w9ysg9ifzy7jcxv0ilpas539gr3i2002-linux-aarch64-unknown-linux-gnu-6.1.21-1.20230405.drv' failed to build
error: 1 dependencies of derivation '/nix/store/niyw27xa52g8zjvd1g0qn81v9hd984zg-nixos-system-pi5-23.11.20240429.576ecd4.drv' failed to build
```
when using `inputs.nixpkgs.url = "nixpkgs/nixos-23.11";`. Does it need the `pkgs` from `inputs.nixpkgs-unstable.url = "nixpkgs/nixos-unstable"?

Reply:
Yes, you need `nixos-unstable`

Reply:
Unstable leads to another issue:

```shell
nix build .\#nixosConfigurations.pi5.config.system.build.toplevel
warning: Git tree '/home/mahene/Programming/nixosconfig' is dirty
error: builder for '/nix/store/8yrxpsgyb8m37bd7sq0dr7likc00pcq4-linux-aarch64-unknown-linux-gnu-6.1.63-stable_20231123-modules-shrunk.drv' failed with exit code 1;
       last 10 log lines:
       >   dependency already copied: /nix/store/x7h0xgyan38f84c0hpm0913z7hvc6xbz-linux-aarch64-unknown-linux-gnu-6.1.63-stable_20231123-modules/lib/modules/6.1.63/kernel/drivers/scsi/scsi_transport_spi.ko.xz
       >   copying dependency: /nix/store/x7h0xgyan38f84c0hpm0913z7hvc6xbz-linux-aarch64-unknown-linux-gnu-6.1.63-stable_20231123-modules/lib/modules/6.1.63/kernel/drivers/message/fusion/mptbase.ko.xz
       >   copying dependency: /nix/store/x7h0xgyan38f84c0hpm0913z7hvc6xbz-linux-aarch64-unknown-linux-gnu-6.1.63-stable_20231123-modules/lib/modules/6.1.63/kernel/drivers/message/fusion/mptscsih.ko.xz
       >   copying dependency: /nix/store/x7h0xgyan38f84c0hpm0913z7hvc6xbz-linux-aarch64-unknown-linux-gnu-6.1.63-stable_20231123-modules/lib/modules/6.1.63/kernel/drivers/message/fusion/mptspi.ko.xz
       > root module: vmxnet3
       >   copying dependency: /nix/store/x7h0xgyan38f84c0hpm0913z7hvc6xbz-linux-aarch64-unknown-linux-gnu-6.1.63-stable_20231123-modules/lib/modules/6.1.63/kernel/drivers/net/vmxnet3/vmxnet3.ko.xz
       > root module: vsock
       >   copying dependency: /nix/store/x7h0xgyan38f84c0hpm0913z7hvc6xbz-linux-aarch64-unknown-linux-gnu-6.1.63-stable_20231123-modules/lib/modules/6.1.63/kernel/net/vmw_vsock/vsock.ko.xz
       > root module: sun4i-drm
       > modprobe: FATAL: Module sun4i-drm not found in directory /nix/store/x7h0xgyan38f84c0hpm0913z7hvc6xbz-linux-aarch64-unknown-linux-gnu-6.1.63-stable_20231123-modules/lib/modules/6.1.63
       For full logs, run 'nix log /nix/store/8yrxpsgyb8m37bd7sq0dr7likc00pcq4-linux-aarch64-unknown-linux-gnu-6.1.63-stable_20231123-modules-shrunk.drv'.
error: 1 dependencies of derivation '/nix/store/qdnl9iypf13y7bm5wrhjpwg2bn2xddls-initrd-linux-aarch64-unknown-linux-gnu-6.1.63-stable_20231123.drv' failed to build
error: 1 dependencies of derivation '/nix/store/qgvfhpzj5kjfa2ac21fi5dvkkc072pd9-nixos-system-pi5-24.05.20240427.58a1abd.drv' failed to build
```

This probably has to do with the pi 4 kernel, on which the Pi 5 is built on. This one shows the same error:

```nix
  imports = [
    nixos-hardware.nixosModules.raspberry-pi-5
    ...
   ];
```

Reply:
Are you including `profiles/all-hardware.nix` (or one of the installer profiles that includes it)? I don't think that module is necessarily expected to be compatible with vendor kernels that might not include the same modules.

Reply:
There's also an existing issue on this repo: #360

Reply:
Okay i got it fixed with the suggestion of https://github.com/NixOS/nixpkgs/issues/126755#issuecomment-869149243:
```nix
nixosConfigurations.pi5 = nixpkgs-unstable.lib.nixosSystem { # Desktop profile
        specialArgs = {
          # contains rpi5 kernel https://github.com/NixOS/nixos-hardware/pull/927/files
          nixos-hardware = nixos-hardware-pi5;
        };
        modules = [
          ./machines/pi5.nix
           ({ pkgs, ... }: {
              nixpkgs = {
                localSystem = {
                  system = "x86_64-linux";
                };
                crossSystem = {
                  system = "aarch64-linux";
                };
               overlays = [   # <-------------------------------
                  (final: super: {
                     makeModulesClosure = x: super.makeModulesClosure (x // { allowMissing = true; });
                   })
               ];
            })
        ];
      };
```

Reply:
Added to `flake.nix`.